### PR TITLE
[TS] LPS-195096 sort by title on Layout

### PR DESCRIPTION
**Pull request by joshuacords**
https://liferay.atlassian.net/browse/LPS-195096

**Author:** @joshuacords 

For all Assets, [AssetEntryDocumentContributor](https://github.com/liferay/liferay-portal/blob/3eb74d2174290686650931d7518263eef612dbfd/modules/apps/asset/asset-service/src/main/java/com/liferay/asset/internal/search/AssetEntryDocumentContributor.java#L111-L116) will add the necessary localized_title_[locale]_sortable fields, but because Layouts aren't assets and don't use Titles, they need to be responsible for adding the field themselves.

### CODE REVIEW

**Comment by joshuacords** [comment]
ci:test:stop

**Comment by joshchong** [comment]
ci:test:sf

**Comment by xbrianlee** [comment]
ci:test:sf

File: modules/apps/layout/layout-test/src/testIntegration/java/com/liferay/layout/search/test/LayoutMultiLanguageSearchTest.java [hunk 1]
```diff
@@ -84,6 +85,24 @@ public void testJapaneseTitle() throws Exception {
...
+	@Test
+	public void testLocalizedFields() throws Exception {
+		Locale locale = LocaleUtil.US;
+
+		setTestLocale(locale);
+
+		_addLayoutMultiLanguage();
+
+		Document document = layoutIndexerFixture.searchOnlyOne(
+			_ENGLISH_KEYWORD, locale);
```
Comment (gustavolimav) [inline modules/apps/layout/layout-test/src/testIntegration/java/com/liferay/layout/search/test/LayoutMultiLanguageSearchTest.java:97]: Trying to preview what Bchan would change maybe it is a good idea to Inline `locale` in both usages?

**Review by gustavolimav — CHANGES_REQUESTED** [review]
Other than the small change commented the code looks good

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search

File: modules/apps/layout/layout-test/src/testIntegration/java/com/liferay/layout/search/test/LayoutMultiLanguageSearchTest.java (hunk 1 above)
Comment (joshuacords) [reply modules/apps/layout/layout-test/src/testIntegration/java/com/liferay/layout/search/test/LayoutMultiLanguageSearchTest.java]: Did it, what do you think now @gustavolimav?

**Comment by joshchong** [comment]
ci:test:sf

**Comment by joshchong** [comment]
ci:test:search

**Comment by ling-alan-huang** [comment]
ci:test:sf

**Comment by joshchong** [comment]
ci:test:relevant

**Comment by joshchong** [comment]
Hey @joshuacords, there are two failing integration tests that look related to this pr. can you take a look?

[LayoutIndexerIndexedFieldsTest.testIndexedFields](https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/753143/testReport/junit/com.liferay.layout.search.test/LayoutIndexerIndexedFieldsTest/testIndexedFields/)

[LayoutMultiLanguageSearchTest.testLocalizedFields](https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/753143/testReport/junit/com.liferay.layout.search.test/LayoutMultiLanguageSearchTest/testLocalizedFields/)

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshchong** [comment]
@joshuacords fyi aborted the ci:test:search run since there is an sf failure.

**Comment by joshuacords** [comment]
Thanks @joshchong, the SF error is still handicapping my machine :frowning_face:

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshchong** [comment]
@joshuacords, hm not sure why. I wonder if you have to run `ant setup-sdk` to `ant compile` to "fetch" the latest fix.

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshchong** [comment]
ci:test:relevant

**Review by gustavolimav — APPROVED** [review]

**Comment by joshchong** [comment]
@joshuacords @gustavolimav Tests look good :+1:. You can manually forward this PR when ready.

**Comment by joshuacords** [comment]
ci:forward:force

**Comment by joshchong** [comment]
sent to BChan: https://github.com/brianchandotcom/liferay-portal/pull/140505

**Pull request closed without merging**